idaes/core/util/utility_minimization.py

```diff
@@ -0,0 +1,674 @@
+    Args:
+        Tin: Inlet temperatures of cooling/heating equipment
+        Toout: Outlet temperatures of cooling/heating equipment
+        Q: Heat of cooling/heating equipment
+    Returns:
+        Tstar: Cumulative Temperature array
+        NQstar: Cumulative heat array
+    """
+    # Q < 0 for hot streams = heat removing = cooling units
+    # Q > 0 for cold streams = heat added = heating units
+
+    # Ignoring edge cases for phase changes
+    # Shaping Temperature array
+    # Calls unique function to avoid  repeating
+    Tstart = unique(np.vstack([Tin, Tout]).reshape((1, -1)))
+    Tstar = np.sort(Tstart[0])
+
+    # Generate vector of Tin size and Tunique
+    nTin = len(Tin)
+    nTstar = len(Tstar)
+    Qmat = np.zeros((nTin, nTstar))
+
+    # Generate cumulative arays for temperature and heat
+    for i in range(nTin):
+        for j in range(nTstar):
+            Qmat[i, j] = linear_interpolation([Tin[i], Tout[i]],
+                                              [0, Q[i]], Tstar[j])
+    Qstar = sum(Qmat, 0)
+    NQstar = Qstar.reshape(nTstar)
+    return Tstar, NQstar
+
+
+def linear_interpolation(x, y, t):
+    """
+    Function to do Linear interpolation with nearest neighbor extrapolation
+    Args:
+        x: Inlet and Outlet Temperature values
+        y: 0 and Heat value
+        t: Unique inlet and outlet temperatures
+    Returns:
+        Qstar: New array of Q Values
+    """
+    # Set the upper or lower value
+    if x[0] < x[1]:
+        lo = 0
+        hi = 1
+    else:
+        lo = 1
+        hi = 0
+    # do Linear interpolation with nearest neighbor extrapolation
+    alpha = (x[hi] - t)/(x[hi] - x[lo])
+    alpha = max([0, alpha])
+    alpha = min([alpha, 1])
+    Qsta = alpha*(y[hi] - y[lo]) + y[lo]
+    return Qsta
+
+
+def print_HX_results(blk, exchanger_list):
+    """
+    Function to print results of Heat Exchangers
+    Args:
+        blk: flowsheet of the equipment
+        exchanger_list: List of equipment to print data
+    Returns:
+        Printed List
+    """
+    # Initialize dictionary and fill with exchanger list
+    exchangerdict = {}
+
+    for i, el in enumerate(exchanger_list):
+        exchangerdict[str(el)] = el
+
+    # initialize null dictionaries for data to be printed
+    Tin_ = {}
+    Tout_ = {}
+    f_ = {}
+    Q_ = {}
+    # Loop over heat exchangers
+    for i in exchangerdict.keys():
+        Tin_[i] = value(exchangerdict[i].control_volume.properties_in[0].temperature)
+        Tout_[i] = value(exchangerdict[i].control_volume.properties_out[0].temperature)
+        f_[i] = value(exchangerdict[i].control_volume.properties_out[0].flow_mol)
+        Q_[i] = value(exchangerdict[i].control_volume.heat[0])
+
+    # Print the header
+    print("Heat Exchanger Summary: ")
+
+    # Print Inlet Temperature, Outlet Temperature and Heat
+    for i in exchangerdict.keys():
+        print("Heat exchanger: ", exchangerdict[i])
+        print('{0:>{1}s}'.format("Inlet T: ", 16),
```

Comment:
Pylint is complaining about the formatting of these print statements - too many arguments for `format`. I think the issue is the `3` in the second `format` method - you don't actually use it in the string formatting ad the 3 is hard-coded there.

I also think you could probably make this code easier to read with something similar to:

```
f"Inlet T:{" "*7} {Tin_[i]:0.3f} K/n"
```